petastorm/spark/spark_dataset_converter.py

```diff
@@ -475,13 +468,22 @@ def _cache_df_or_retrieve_cache_data_url(df, parent_cache_dir_url,
                     meta.dtype == dtype:
                 return meta.cache_dir_url
         # do not find cached dataframe, start materializing.
-        cached_df_meta = CachedDataFrameMeta.create_cached_dataframe(
+        cached_df_meta = CachedDataFrameMeta.create_cached_dataframe_meta(
             df, parent_cache_dir_url, parquet_row_group_size_bytes,
             compression_codec, dtype)
         _cache_df_meta_list.append(cached_df_meta)
         return cached_df_meta.cache_dir_url
 
 
+def _remove_df_from_cache(cache_dir_url):
```

Comment:
What about we call it `_remove_cache_metadata_and_data(...)` and rename `_delete_dir_handler` as `_remove_cache_data_handler`? It would be less confusing which function is doing what, and also match the semantics of `delete_cache_data_atexit` (basically we call the underlying files "data" and the in-memory structure "metadata").

Reply:
`_remove_cache_metadata_and_data` is good.
keep `_delete_dir_handler` for it can be used in other places when deleting dir.